core/src/test/java/org/elasticsearch/script/ScriptMetaDataTests.java

```diff
@@ -149,6 +150,17 @@ protected ScriptMetaData createTestInstance() {
     }
 
     @Override
+    protected MutateFunction<ScriptMetaData> getMutateFunction() {
+        return instance -> {
+            // ScriptMetaData doesn't allow us to see the scripts inside it so
+            // the best we can do here is create a new random instance and rely
+            // on the fact that the new instance is very unlikely to be equal to
+            // the old one
+            return createTestInstance();
```

Comment:
You are right on this, good point. using equals to do the assume was kind of a silly idea :)